lib/galaxy/jobs/runners/util/cli/job/lsf.py

```diff
@@ -0,0 +1,102 @@
+# A simple CLI runner for slurm that can be used when running Galaxy from a
+# non-submit host and using a Slurm cluster.
+from logging import getLogger
+
+try:
+    from galaxy.model import Job
+    job_states = Job.states
+except ImportError:
+    # Not in Galaxy, map Galaxy job states to Pulsar ones.
+    from pulsar.util import enum
+    job_states = enum(RUNNING='running', OK='complete', QUEUED='queued', ERROR="failed")
+
+from ..job import BaseJobExec
+
+log = getLogger(__name__)
+
+argmap = {
+    'memory': '-M',
+    'cores': '-n',
+    'queue': '-q',
+    'working_dir': '-cwd'
+}
+
+
+class LSF(BaseJobExec):
+
+    def __init__(self, **params):
+        self.params = {}
+        for k, v in params.items():
+            self.params[k] = v
+
+    def job_script_kwargs(self, ofile, efile, job_name):
+        scriptargs = {'-o': ofile,
+                      '-e': efile,
+                      '-J': job_name}
+
+        # Map arguments using argmap.
+        for k, v in self.params.items():
+            if k == 'plugin':
+                continue
+            try:
+                if not k.startswith('-'):
+                    k = argmap[k]
+                if k == 'memory':
+                    # Memory requires both -m and -R rusage[mem=v] request
+                    scriptargs['-R'] = "\"rusage[mem=%s]\"" % v
+                scriptargs[k] = v
+            except Exception:
+                log.warning('Unrecognized long argument passed to LSF CLI plugin: %s' % k)
+
+        # Generated template.
+        template_scriptargs = ''
+        for k, v in scriptargs.items():
+            template_scriptargs += '#BSUB %s %s\n' % (k, v)
+        return dict(headers=template_scriptargs)
+
+    def submit(self, script_file):
+        return 'bsub <%s' % script_file
+
+    def delete(self, job_id):
+        return 'bstop %s' % job_id
+
+    def get_status(self, job_ids=None):
+        return "bjobs -o \"id stat delimiter='^'\" -noheader"  # check this
+
+
+    def get_single_status(self, job_id):
+        return "bjobs -o stat -noheader " + job_id
+
+    def parse_status(self, status, job_ids):
+        # Get status for each job, skipping header.
+        rval = {}
+        for line in status.splitlines()[1:]:
+            id, state = line.split('^')
+            if id in job_ids:
+                # map job states to Galaxy job states.
+                rval[id] = self._get_job_state(state)
+        return rval
+
+    def parse_single_status(self, status, job_id):
+        if "is not found" in status:
+            raise Exception("Job %s not found in LSF" % job_id)
```

Comment:
Not sure if I should be handling the 'not found' case here though through an exception. Please advise.

Reply:
The other plugins assume the job has completed and is no longer being tracked by the DRM:

```
        # no state found, job has exited
        return job_states.OK
```

Stick with that? If you are confident the LSF is going to reserve and track that job_id for a long time and so this indicates something definitely went wrong you could return job_states.ERROR instead - this would be better than throwing an exception I think (this is used in a loop that doesn't catch and handle that exception https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/jobs/runners/cli.py#L154).

Reply:
I guess you are right, after many years of not using LSF, what I noticed now is that it only keeps the id IFF the job fails. I think I have added -a as well, which will show all jobs as well (at the expense maybe of showing too many jobs). Will change to the behaviour that you suggest.